### PR TITLE
For php7.2

### PR DESCRIPTION
**Pull request by okashitay**
# Overview

Issue https://github.com/apereo/phpCAS/issues/255

Pull Request OK?
This pull request works with php 7.2 or higher.

Please use ver.1.3.5 for PHP 7.1 and below.

### CODE REVIEW

**Comment by okashitay** [comment]
Sorry, I was not reading the document 
https://wiki.jasig.org/display/CASC/Developing+phpCAS

I will read it now.

**Comment by adamfranco** [comment]
Hi @okashitay , thanks for starting on this PR! One thing I'm a bit unclear about is which changes are 7.2-only and break usage on PHP 7.1. I know the latest Red Hat Enterprise Linux 7 and CentOS 7 are still shipping with PHP 7.1, so while it will be be worth breaking backward compatibility with PHP 5.x soon, I'd rather not break BC with PHP 7.1 unless there are significant demonstrable benefits.

**Comment by okashitay** [comment]
Hi @adamfranco , 
thank you for the message.
We will respond so that it can operate even in PHP 7.1.
Please wait for a while.

**Comment by doMynation** [comment]
Hey, firstly, thanks to all the contributors on this project. We've been using this library for years where I work.

If you don't mind me asking, what's the status for this PR? Has it been abandoned? We're migrating to PHP 7.2 soon and we're worried that phpCAS might be a blocker.

**Comment by phy25** [comment]
> Hey, firstly, thanks to all the contributors on this project. We've been using this library for years where I work.
> 
> If you don't mind me asking, what's the status for this PR? Has it been abandoned? We're migrating to PHP 7.2 soon and we're worried that phpCAS might be a blocker. I couldn't find a mention of PHP7+ compatibility in the documentation. If anyone could give me some pointers I'd greatly appreciate.

Hi @doMynation are you experiencing issues in PHP 7.2? Please see #255 because it seems that compatibility issue has been resolved.

**Comment by doMynation** [comment]
> Hi @doMynation are you experiencing issues in PHP 7.2? Please see #255 because it seems that compatibility issue has been resolved.

Thank you for the prompt reply. I am not experiencing issues yet, I was mostly wondering (for my own peace of mind) if PHP 7.2 was officially supported.

File: test/CAS/Tests/Cas20AttributesTest.php [hunk 1]
```diff
@@ -37,7 +37,12 @@
  * @license  http://www.apache.org/licenses/LICENSE-2.0  Apache License 2.0
  * @link     https://wiki.jasig.org/display/CASC/phpCAS
  */
-class CAS_Tests_Cas20AttributesTest extends PHPUnit_Framework_TestCase
+
+require_once dirname(__FILE__) ."/../../../vendor/autoload.php";
```
Comment (phy25) [inline test/CAS/Tests/Cas20AttributesTest.php]: This autoload require is not necessary?

Comment (phy25) [inline test/CAS/Tests/Cas20AttributesTest.php]: These autoload `require`s are not necessary?

File: composer.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
         "ext-curl": "*"
     },
     "require-dev": {
-        "phpunit/phpunit": "~3.7.10"
+        "phpunit/phpunit": "~6.5.7"
```
Comment (phy25) [inline composer.json]: Is PHPUnit 7 better? (PHPUnit 8 only supports PHP7+, which is not good here)

**Comment by jdufresne** [comment]
Rather than maintaining `Autoload.php`, is there any interest by the project maintainers in moving towards the [PSR-4 autoloading](https://www.php-fig.org/psr/psr-4/) convention instead?

If there is, I could give it a shot and it would move the package towards modern PHP community idioms.

**Comment by phy25** [comment]
> Rather than maintaining `Autoload.php`, is there any interest by the project maintainers in moving towards the [PSR-4 autoloading](https://www.php-fig.org/psr/psr-4/) convention instead?

I would say this will need to be eventually implemented at some point in the future, but we just need to retain compatibility somehow..?

**Comment by phy25** [comment]
Personally we prefer composer now, and this work is also covered by #332. I am going to close the PR.

Thank you for raising the issue and attempting to fix it though!

**Pull request closed without merging**